kyndryl-design-system/shidoka-applications
# fix(form components): : dark mode token updates

## Summary

Implement Dark Mode color tokens from updated Figma files for 2.0 Breaking Changes on the following components:

- [x] Checkbox -- (* check after Robert's new tokens are updated, may need to update checked base tokens)
- [x] Radio Button -- (* check after Robert's new tokens are updated)
- [x] Text Input
- [x] Text Area
- [x] Toggle Button
- [x] Dropdown -- (* check tags when tokens updated)
- [x] Number Input (* check +/- when button tokens updated)
- [x] Form Label

- [x] `form-input.scss`* (base styles for most form input components) -- check WARNING, check whether disabled labels should have styling

Form components that will be handled on a separate PR:
- [Date Picker](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003589)
- [Date Range Picker](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003590)
- [Timepicker](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003569)

## ADO Story or GitHub Issue Link

- [Checkbox](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003593)
- [Text Input](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003570)
- [Text Area](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003571)
- [Toggle Button](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003568)
- [Dropdown](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003587)
- [Number Input](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003582)
- [Radio Button](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003577)

## Figma Links

- [Checkbox](https://www.figma.com/design/CQuDZEeLiuGiALvCWjAKlu/Applications---Component-Library?node-id=13336-80710&node-type=section&m=dev)
- [Text Input](https://www.figma.com/design/CQuDZEeLiuGiALvCWjAKlu/Applications---Component-Library?node-id=4045-141810&node-type=canvas&m=dev)
- [Text Area](https://www.figma.com/design/CQuDZEeLiuGiALvCWjAKlu/Applications---Component-Library?node-id=4285-236267&node-type=canvas&m=dev)
- [Toggle Button](https://www.figma.com/design/CQuDZEeLiuGiALvCWjAKlu/Applications---Component-Library?node-id=4211-233454&node-type=instance&m=dev)
- [Dropdown](https://www.figma.com/design/CQuDZEeLiuGiALvCWjAKlu/Applications---Component-Library?node-id=4161-101206&node-type=instance&m=dev)
- [Number Input](https://www.figma.com/design/CQuDZEeLiuGiALvCWjAKlu/Applications---Component-Library?node-id=14893-233109&node-type=canvas&m=dev)
- [Radio Button](https://www.figma.com/design/CQuDZEeLiuGiALvCWjAKlu/Applications---Component-Library?node-id=4193-260552&node-type=symbol&m=dev)

## Checklist


## Figma/Storybook Inconsistencies

- ~~thumb color on toggleButton hover in dark mode (`--kd-color-background-ui-hollow-default`):~~
- `Background/UI/Default/Secondary` (`--kd-color-background-ui-default-secondary`) -- not available in storybook/`tokens` (push requested from design team)
- ~~`Background/Container/Default` (`--kd-color-background-container-default`) doesn't match figma~~

## Testing Instructions
- all of these components should have a Storybook sidenav item, except for Form Label, that can be most easily checked in the `Patterns/Forms` story
- there are still a few inconsistencies in the colors as they appear in Figma vs. Storybook. i've noted all that i could find in the section above, but please make any notes of others that you might see. the tokens themselves should all be consistent between the code and the figma file, just not fully reflected yet in storybook

## Screenshots

(if any)
